modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java

```diff
@@ -872,23 +876,234 @@ public void testBindPublishedObjectDefinitions() throws Exception {
 
 		// Bind two published object definitions
 
+		ObjectDefinition objectDefinitionB =
+			_addAndPublishCustomObjectDefinition("B");
+		ObjectDefinition objectDefinitionBB =
+			_addAndPublishCustomObjectDefinition("BB");
+
 		_testBindObjectDefinitions(
-			_addAndPublishCustomObjectDefinition("A"),
-			_addAndPublishCustomObjectDefinition("AA"),
+			objectDefinitionB, objectDefinitionBB,
 			(objectDefinition1, objectDefinition2) ->
 				TreeTestUtil.assertObjectDefinitionTree(
 					LinkedHashMapBuilder.put(
-						"A", new String[] {"AA"}
+						"B", new String[] {"BB"}
 					).put(
-						"AA", new String[0]
+						"BB", new String[0]
 					).build(),
 					_treeFactory.createObjectDefinitionTree(
 						objectDefinition1.getObjectDefinitionId(),
 						_objectDefinitionLocalService::getObjectDefinition),
 					_objectDefinitionLocalService));
 
+		// The object definitions must have the same scope
+
+		ObjectDefinition siteObjectDefinition =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				Collections.singletonList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"a" + RandomTestUtil.randomString()
+					).build()),
+				ObjectDefinitionConstants.SCOPE_SITE);
+
+		AssertUtils.assertFailure(
+			ObjectRelationshipEdgeException.class,
+			"Unable to bind object definitions when they have different scopes",
+			() -> _bindObjectDefinitions(
+				objectDefinitionB.getObjectDefinitionId(),
+				siteObjectDefinition.getObjectDefinitionId()));
+
+		// The object relationship must not create a circular reference in
+		// a root context
```

Comment:
Following the same principle, what do you think about renaming to:
// Unable to bind the object definitions because the object relationship must not create a circular reference in a root context